Shipfinity.Api/Controllers/ProductController.cs

```diff
@@ -158,5 +160,42 @@ private async Task<IActionResult> GetProductsInRange(int skip, int take)
                 return BadRequest(ex.Message);
             }
         }
+        
+        [HttpPost("{productId}/UploadPhoto")]
+        public async Task<IActionResult> UploadPhoto(int productId, IFormFile file)
+        {
+            if (file == null || file.Length == 0)
+            {
+                return BadRequest("No file uploaded.");
+            }
+
+            try
+            {
+                // Ensure the images directory exists
+                if (!Directory.Exists(_imagesPath))
+                {
+                    Directory.CreateDirectory(_imagesPath);
+                }
+
+                // Generate a unique file name to avoid overwriting existing files
+                var fileName = Guid.NewGuid() + Path.GetExtension(file.FileName);
+                var filePath = Path.Combine(_imagesPath, fileName);
+
+                // Save the file
+                using (var stream = new FileStream(filePath, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+
+                // Update product's photo URL in the database
+                await _productService.UpdateProductPhotoUrl(productId, "/images/" + fileName);
+
+                return Ok("File uploaded successfully.");
+            }
+            catch (Exception ex)
```

Comment:
W need to separately catch the product not found exception and return 400 bad request here

Reply:
added exceptipon